### PR TITLE
Jina: update secrets management

### PR DESCRIPTION
**Pull request by anakin87**
part of #366

### CODE REVIEW

File: integrations/jina/src/haystack_integrations/components/embedders/jina/text_embedder.py [hunk 1]
```diff
@@ -48,22 +48,22 @@ def __init__(
...
         """
 
-        api_key = api_key or os.environ.get("JINA_API_KEY")
-        # we check whether api_key is None or an empty string
-        if not api_key:
+        resolved_api_key = api_key.resolve_value()
+        if resolved_api_key is None:
             msg = (
-                "JinaTextEmbedder expects an API key. "
+                "JinaDocumentEmbedder expects an API key. "
```
Comment (julian-risch) [inline integrations/jina/src/haystack_integrations/components/embedders/jina/text_embedder.py:54]: Needs to be changed back to `JinaTextEmbedder`

**Review by julian-risch — CHANGES_REQUESTED** [review]
One copy-paste issue that needs to be fixed. Otherwise looks good to me. 👍

File: integrations/jina/src/haystack_integrations/components/embedders/jina/text_embedder.py (hunk 1 above)
Comment (anakin87) [reply integrations/jina/src/haystack_integrations/components/embedders/jina/text_embedder.py]: fixed. Thanks!

**Review by julian-risch — APPROVED** [review]
LGTM! 👍

File: integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py [hunk 2]
```diff
@@ -57,16 +56,15 @@ def __init__(
...
-        api_key = api_key or os.environ.get("JINA_API_KEY")
-        # we check whether api_key is None or an empty string
-        if not api_key:
+        resolved_api_key = api_key.resolve_value()
+        if resolved_api_key is None:
             msg = (
                 "JinaDocumentEmbedder expects an API key. "
                 "Set the JINA_API_KEY environment variable (recommended) or pass it explicitly."
             )
             raise ValueError(msg)
```
Comment (shadeMe) [inline integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py:65]: `resolve_value()` will raise an error if `strict=True`, which it is by default, and the env var is not set, so this is dead code. Let's remove it and directly call the method when updating the headers. Same for the other component too.

**Pull request merged**